component/src/main/java/org/wso2/extension/siddhi/store/influxdb/InfluxDBEventTable.java

```
@@ -0,0 +1,733 @@
+        } else {
+            selectQuery.append(INFLUXQL_WHERE);
+            for (Map.Entry<String, Object> entry : parameterMap.entrySet()) {
+                Object myObject = entry.getValue();
+                if (myObject instanceof String) {
+                    condition = condition.replaceFirst(Pattern.quote("?"), myObject.toString());
+                } else {
+                    condition = condition.replaceFirst(Pattern.quote("'?'"), myObject.toString());
+                }
+            }
+            for (Map.Entry<Integer, Object> entry : constantMap.entrySet()) {
+                Object myObject = entry.getValue();
+                if (myObject instanceof String) {
+                    condition = condition.replaceFirst(Pattern.quote("*"), myObject.toString());
+                } else {
+                    condition = condition.replaceFirst(Pattern.quote("'*'"), myObject.toString());
+                }
+            }
+            selectQuery.append(condition);
+        }
+        InfluxDBCompiledCondition compiledGroupByClause = influxDBCompiledSelection.getCompiledGroupByClause();
+        if (compiledGroupByClause != null) {
+            String groupByClause = "group by " + compiledGroupByClause.getCompiledQuery();
+            selectQuery.append(WHITESPACE).append(groupByClause);
+        }
+        if (influxDBCompiledSelection.getLimit() != null) {
+            selectQuery.append(" limit ").append(influxDBCompiledSelection.getLimit());
+        }
+        if (influxDBCompiledSelection.getOffset() != null) {
+            selectQuery.append(" offset ").append(influxDBCompiledSelection.getOffset());
+        }
+        InfluxDBCompiledCondition compiledHavingClause = influxDBCompiledSelection.getCompiledHavingClause();
+        if (compiledHavingClause != null) {
+            log.error("Having clause is defined in the query for " + this.tableName +
+                    " But it is not defined for influxDB store implementation  ");
```

Instead of logging, throw a `InfluxDBTableException` here.